File: mymixtape-app/src/types/api/response.ts
export type ApiResponseData = {};

export type ApiResponse = {
  headers?: Headers;
  data: ApiResponseData;
};

export interface AuthorizationUrlResponse extends ApiResponse {
  data: {
    url: string;
  };
}

export interface AccessTokenResponse extends ApiResponse {
  data: {};
}

export interface UserProfileResponse extends ApiResponse {
  data: {
    id: string;
    name: string;
    images: {
      url: string;
      height: number;
      width: number;
    }[];
  };
}

export interface UserPlaylistsResponse extends ApiResponse {
  data: {
    href: string;
    limit: number;
    next: string;
    offset: number;
    previous: string;
    total: number;
    items: {
      id: string;
      name: string;
      images: {
        url: string;
        height: number;
        width: number;
      }[];
    }[];
  };
}

export interface CombinePlaylistResponse extends ApiResponse {
  data: {
    id: string;
    name: string;
  };
}

export interface TestProfileResponse extends ApiResponse {
  data: {
    count: number;
  };
}

export interface GetSessionResponse extends ApiResponse {
  data: {
    count: number;
  };
}

export interface GetSessionTokenResponse extends ApiResponse {
  data: {
    token: string;
  };
}

export interface SetSessionTokenResponse extends ApiResponse {
  data: {
    token: string;
  };
}

// Error Response
export interface ErrorResponse extends ApiResponse {
  status: number;
  message: string;
}

File: mymixtape-api/models/response.go
package models

// Authorization
type AuthorizationUrlResponse struct {
	Url		string	`json:"url"`
	ValidToken	bool	`json:"valid_token"`
}

type AccessTokenResponse struct {
	Token	string	`json:"token"`
	ExpiresIn	int	`json:"expires_in"`
}

// User Profile
type CurrentUsersProfileResponse struct {
	ID		string	`json:"id"`
	Name	string	`json:"name"`
	Images []struct {
		URL    string `json:"url"`
		Height int    `json:"height"`
		Width  int    `json:"width"`
	} `json:"images"`
}


// User Playlists
type CurrentUsersPlaylistsResponse struct {
	Href	string	`json:"href"`
	Limit	int		`json:"limit"`
	Next	string	`json:"next"`
	Offset	int		`json:"offset"`
	Previous	string	`json:"previous"`
	Total	int		`json:"total"`
	Items    []struct {
		ID     string `json:"id"`
		Images []struct {
			URL    string `json:"url"`
			Height int    `json:"height"`
			Width  int    `json:"width"`
		} `json:"images"`
		Name  string `json:"name"`
	} `json:"items"`
}

type PlaylistItemsResponse struct {
	Items    []struct {
		Track   struct {
			ID         string `json:"id"`
			Name        string `json:"name"`
			Type        string `json:"type"`
		} `json:"track"`
	} `json:"items"`
}

type CombinePlaylistResponse struct {
	ID	string	`json:"id"`
	Name	string	`json:"name"`
}

// Error Response
type ErrorResponse struct {
	Message	string	`json:"message"`
	Status	int		`json:"status"`
}